### PR TITLE
executor: display Cardinality info from stats for `SHOW INDEX` | tidb-test=pr/2107

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/42227

Problem Summary:

### What is changed and how it works?

we didn't display the NDV information in the `SHOW INDEX` statement.

This pr intends to add it.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by time-and-fate — APPROVED** [review]

File: executor/show.go [hunk 1]
```diff
@@ -739,14 +744,19 @@ func (e *ShowExec) fetchShowIndex() error {
 				break
 			}
 		}
+		colStats, ok := statsTbl.Columns[pkCol.ID]
+		var ndv int64
+		if ok {
```
[the remark this answers is not in the document]
Comment (winoros) [reply executor/show.go:749]: I think it's a rare problem. Just missed the comment this time.

**Review by fixdb — APPROVED** [review]
+1

**Pull request merged**